init/jstParser.js

```diff
@@ -0,0 +1,81 @@
+class JstParser {
+  constructor(jst, settings = {}) {
+    this.jst = jst;
+    let { params, conditions, loops } = settings;
+    !conditions ? (this.conditions = []) : (this.conditions = conditions);
```

Comment:
```suggestion
    this.conditions = conditions || [];
```
Refactor this patter everywhere